lang/LangPrimSource/PyrListPrim.cpp

```diff
@@ -80,7 +80,9 @@ int prArrayMultiChanExpand(struct VMGlobals *g, int numArgsPushed)
 	}
 
 	obj2 = newPyrArray(g->gc, maxlen, 0, true);
-	SetObject(a, obj2);
+	// push obj2 onto the stack so it's not collected
+	// we can't just set in a here as we still need obj1's slots below, so we need to ensure it isn't collected
+	++g->sp; SetObject(g->sp, obj2);
```

Comment:
> when/how can the new object be collected mid call? (naive question)

If you replace the object on the stack, and it is not reachable otherwise (held in an object or a var), and then you make a new object, it could happen. So fun!

Reply:
> also, please place each statement on a separate lines (here and elsewhere)

yeah sorry. Copied style form elsewhere in the code base.

Reply:
> If you replace the object on the stack, and it is not reachable otherwise (held in an object or a var), and then you make a new object, it could happen. So fun!

@muellmusik got it, thanks!